options/options.go

```diff
@@ -0,0 +1,5 @@
+// Package options provides generic functional options pattern utilities.
```

Comment:
this package kinda threw me off :D 
I am familiar with generics but would love to know how we benefit from it. Is this just for DRY?